src/Microsoft.Extensions.Configuration.AzureAppConfiguration/AzureAppConfigurationProvider.cs

```diff
@@ -260,112 +278,112 @@ public async Task RefreshAsync(CancellationToken cancellationToken)
+                                        refreshableFfWatchers,
+                                        watchedFfCollections,
+                                        endpoint,
+                                        logDebugBuilder,
+                                        logInfoBuilder,
+                                        cancellationToken).ConfigureAwait(false);
                                 }
                                 else
                                 {
-                                    logDebugBuilder.AppendLine(LogHelper.BuildKeyValueReadMessage(change.ChangeType, change.Key, change.Label, endpoint.ToString()));
+                                    // Trigger a single load-all operation if a change was detected in one or more key-values with refreshAll: true
+                                    data = new Dictionary<string, ConfigurationSetting>(StringComparer.OrdinalIgnoreCase);
+                                    watchedKvCollections = await LoadSelected(client, data, _options.KeyValueSelectors, cancellationToken).ConfigureAwait(false);
+                                    watchedFfCollections = await LoadSelected(client, data, _options.FeatureFlagSelectors, cancellationToken).ConfigureAwait(false);
+                                    watchedSentinelKvs = await LoadKeyValuesRegisteredForRefresh(client, data, cancellationToken).ConfigureAwait(false);
+                                    logInfoBuilder.AppendLine(LogHelper.BuildConfigurationUpdatedMessage());
+                                    return;
                                 }
                             }
-
-                            if (refreshAll)
+                            else
                             {
-                                // Trigger a single load-all operation if a change was detected in one or more key-values with refreshAll: true
-                                data = await LoadSelectedKeyValues(client, cancellationToken).ConfigureAwait(false);
-                                watchedSettings = await LoadKeyValuesRegisteredForRefresh(client, data, cancellationToken).ConfigureAwait(false);
-                                watchedSettings = UpdateWatchedKeyValueCollections(watchedSettings, data);
-                                logInfoBuilder.AppendLine(LogHelper.BuildConfigurationUpdatedMessage());
-                                return;
-                            }
+                                // Get key value collection changes if RegisterAll was called
+                                watchedKvCollections = new Dictionary<KeyValueSelector, IEnumerable<MatchConditions>>(_watchedKvCollections);
 
-                            changedKeyValuesCollection = await GetRefreshedKeyValueCollections(refreshableMultiKeyWatchers, client, logDebugBuilder, logInfoBuilder, endpoint, cancellationToken).ConfigureAwait(false);
+                                if (_options.RegisterAllEnabled && registerAllIsRefreshable)
```

Comment:
The `_options.RegisterAllEnabled` is redundant here as it has fallen into the else branch(in line 290, you have checked `if(!_options.RegisterAllEnabled)`

Reply:
Missed this when reordering everything, thanks